internal/app/admin/http/handler.go

```diff
@@ -109,52 +109,67 @@ func configDumpHandler(bootstrapConfig *bootstrapv1.Bootstrap) http.HandlerFunc
 	}
 }
 
-// TODO(lisalu): Support dump of matching resources when cache key regex is provided.
+func printCacheEntries(keys []string, cache cache.ReadOnlyCache, w http.ResponseWriter) {
+	resp := marshallableCache{}
+	for _, key := range keys {
+		resource, err := cache.FetchReadOnly(key)
+		if err != nil {
+			w.WriteHeader(http.StatusNotFound)
+			fmt.Fprintf(w, "no resource for key %s found in cache.\n", key)
+			continue
```

Comment:
For public prosperity, we chatted offline and decided to skip the key but not return 404 or log an error. Despite multiple reads this should be OK since it's all an atomic action for this admin API.